pkg/converters/ingress/ingress.go

```diff
@@ -948,6 +963,50 @@ func (c *converter) syncBackendEndpointCookies(backend *hatypes.Backend) {
 	}
 }
 
+func (c *converter) syncBackendEndpointHashes(backend *hatypes.Backend) {
+	if !backend.Server.AssignID {
+		return
+	}
+
+	// We need to take the endpoints in some order to resolve hash collisions... TargetRef will do
+	eps := make([]*hatypes.Endpoint, len(backend.Endpoints))
+	copy(eps, backend.Endpoints)
+	sort.SliceStable(eps, func(i, j int) bool {
+		ep1 := eps[i]
+		ep2 := eps[j]
+		return ep1.TargetRef < ep2.TargetRef
+	})
+
+	usedPUIDS := map[uint32]struct{}{}
+	for _, ep := range eps {
+		if ep.TargetRef == "" {
+			continue
+		}
+
+		var hash uint32
+		pod, err := c.cache.GetPod(ep.TargetRef)
+		if err == nil {
+			hasher := fnv.New32a()
+			hasher.Write([]byte(pod.UID))
+			// We get a uint32, but haproxy uses an int32 and insists on it being nonnegative,
+			// so truncate to 31 bits.
+			hash = hasher.Sum32() & 0x7fffffff
+		} else {
+			hash = 1
+			c.logger.Error("error calculating hash value for pod %s; ID assignment won't be stable: %v", ep.TargetRef, err)
+		}
+		for { // If the ID is already used, linearly probe to find one that's not
+			_, exists := usedPUIDS[hash]
+			if hash != 0 && !exists {
+				break
+			}
```

Comment:
what's the meaning of `hash != 0`, maybe you meant `hash != 1` which means error calling `GetPod()`?

Reply:
No, `0` is an invalid value for `id` in HAProxy (conveniently it also gives us a value to use for "don't set the ID" in the template). So if we have to linearly probe forward from 2,147,483,647, we should go to 1, not 0.

Reply:
Commented.